tests/acceptance/test_update.py

```diff
@@ -397,25 +397,145 @@ def test_network_based_image_update(self, successful_image_update_mender, bitbak
+                                            checksum_ok=True,
+                                            header_checksum_ok=True,
+                                            update_written=False,
+                                            artifact_version=3,
+                                            success=False),
+                              SignatureCase(label="EC, Correctly signed, key not present, version 3",
+                                            signature=True,
+                                            signature_ok=True,
+                                            key=False,
+                                            key_type="EC",
+                                            checksum_ok=True,
+                                            header_checksum_ok=True,
+                                            update_written=True,
+                                            artifact_version=3,
+                                            success=True),
+                              SignatureCase(label="EC, Not signed, key present, version 3",
+                                            signature=False,
+                                            signature_ok=False,
+                                            key=True,
+                                            key_type="EC",
+                                            checksum_ok=True,
+                                            header_checksum_ok=True,
+                                            update_written=False,
+                                            artifact_version=3,
+                                            success=False),
+                              SignatureCase(label="EC, Correctly signed, but checksum wrong, key present, version 3",
+                                            signature=True,
+                                            signature_ok=True,
+                                            key=True,
+                                            key_type="EC",
+                                            checksum_ok=False,
+                                            header_checksum_ok=True,
+                                            update_written=True,
+                                            artifact_version=3,
+                                            success=False),
+                              SignatureCase(label="EC, Correctly signed, but header does not match checksum, key present, version 3",
+                                            signature=True,
+                                            signature_ok=True,
+                                            key=True,
+                                            key_type="EC",
+                                            checksum_ok=True,
+                                            header_checksum_ok=False,
+                                            update_written=False,
+                                            artifact_version=3,
```

Comment:
I don't see you using `artifact_version` anywhere. Is that coming?

Reply:
https://github.com/mendersoftware/meta-mender/pull/626/files/47db16c6eceecf22bd305b3a811bc0cf035da76d#diff-379340f30fc917e8f0c80a32964a52b9R575

This will generate an artifact-version 3, if I'm not mistaken?

Reply:
```
       # Generate artifact with specific version. None means default.
         if sig_case.artifact_version is not None:
             artifact_args += " -v %d" % sig_case.artifact_version
```